### PR TITLE
Users/asuciu/recipe details page

### PR DESCRIPTION
**Pull request by suciuandra**
# Justification

#46 

# Implementation

Added recipe details page, create recipe button on recipes page and updated scroll style.

### CODE REVIEW

File: iChef-WebUi/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 import {LandingPageComponent} from './components/landing-page/landing-page.component';
+import { RecipeDetailsComponent } from './components/recipe-details/recipe-details.component';
 import { RecipesComponent } from './components/recipes/recipes.component';
 
 const routes: Routes = [
     {path: '', redirectTo: '/home', pathMatch: 'full'},
     {path: 'home', component: LandingPageComponent},
-    {path: 'recipes', component: RecipesComponent}
+    {path: 'recipes', component: RecipesComponent},
+    {path: 'recipes/details', component: RecipeDetailsComponent}
```
Comment (EnsyFane) [inline iChef-WebUi/src/app/app-routing.module.ts:11]: Can we have the id of the recipe in the url? We need this to be able to share links.

File: iChef-WebUi/src/app/data-types/recipe.ts [hunk 2]
```diff
@@ -1,18 +1,35 @@
 export class Recipe {
+    id?: number;
```
Comment (EnsyFane) [inline iChef-WebUi/src/app/data-types/recipe.ts:2]: id is going to be a string. in the backend we will store it as a GUID

Comment (EnsyFane) [reply iChef-WebUi/src/app/data-types/recipe.ts:2]: all ids are GUIDs

File: iChef-WebUi/src/app/styles/mixins.scss [hunk 3]
```diff
@@ -24,3 +24,38 @@
     border: 3px solid $primary-color;
     border-radius: 25px;
 }
+
+@mixin split-horizontal {
+
+    height: 85%;
```
Comment (EnsyFane) [inline iChef-WebUi/src/app/styles/mixins.scss:30]: remove unnecessary white lines

**Review by EnsyFane — APPROVED** [review]

**Comment by EnsyFane** [comment]
Also, post screenshots of the page

File: iChef-WebUi/src/app/styles/variables.scss [hunk 4]
```diff
@@ -5,10 +5,14 @@ App color palette
  */
 
 $primary-color: #E98074;
+$primary-color2: rgba($color:#E98074, $alpha:0.75);
```
Comment (PinkyPotrick) [inline iChef-WebUi/src/app/styles/variables.scss:8]: I don't think that 'primary-color2' is a good name, maybe rename it to 'primary-color-faded' or other specific term. 
Update: You used the term 'transparent' below, that could work as well.

**Review by PinkyPotrick — CHANGES_REQUESTED** [review]

**Review by PinkyPotrick — APPROVED** [review]

**Review by mirunatodea — APPROVED** [review]

**Pull request merged**